app/assets/javascripts/modules/all-content-finder.js

```diff
@@ -0,0 +1,53 @@
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  'use strict'
+
+  // Enhances the all content finder with dynamic behaviour and analytics tracking
+  class AllContentFinder {
```

Comment:
I did wonder about whether to question ES6 or not here, but I notice we're using Element.closest later which is modern JS without a polyfill so I guess we're already there with modern JS.

Reply:
Yeah – figured we might as well stick with ES6 for this as a "new" piece of functionality. Nothing in this is (or will be as far as I can tell) on the critical path UX-wise, it's just progressive enhancement and analytics so I don't feel too bad about legacy browsers.